counterpartylib/test/conftest.py

```diff
@@ -126,20 +137,139 @@ def cp_server(request):
+
+        if unconfirmed == False:
+            unspent_txouts = filter(lambda txout: txout['confirmations'] > 0, unspent_txouts)
+
+        if multisig_inputs:
+            raise NotImplementedError("%s" % multisig_inputs)
+
+        if unspent_tx_hash:
+            unspent_txouts = filter(lambda txout: txout['txid'] == unspent_tx_hash, unspent_txouts)
+
+        return list(unspent_txouts)
+
+    def update_utxo_set(self, txins, txouts):
+        # spent the UTXOs
+        for txin in txins:
+            self.spent_utxos.append((txin['txid'], txin['vout']))
+
+        for txout in txouts:
+            self.txouts.append({
+                'address': txout['address'],
+                'amount': txout['amount'],
+                'confirmations': int(txout['confirmations']),
+                'scriptPubKey': txout['scriptPubKey'],
+                'txhex': txout['txhex'],
+                'txid': txout['txid'],
+                'vout': txout['vout'],
+            })
+
+    def increment_confirmations(self):
+        for utxo in self.txouts:
+            utxo['confirmations'] = (utxo['confirmations'] or 0) + 1
+
+    def add_raw_transaction(self, raw_transaction, tx_id=None, confirmations=0):
+        tx = pycoin.tx.Tx.from_hex(raw_transaction)
+        tx_id = tx_id or tx.id()
+
+        txins = []
+        for txin in tx.txs_in:
+            txins.append({
+                'txid': pycoin.serialize.b2h_rev(txin.previous_hash),
+                'vout': txin.previous_index
+            })
+
+        txouts = []
+        for idx, txout in enumerate(tx.txs_out):
+            txouts.append({
+                'address': script.scriptpubkey_to_address(bitcoinlib.core.CScript(txout.script)),
+                'confirmations': int(confirmations),
+                'amount': txout.coin_value / 1e8,
+                'scriptPubKey': binascii.hexlify(txout.script).decode('ascii'),
+                'txhex': raw_transaction,
+                'txid': tx_id,
+                'vout': idx
+            })
+
+        # logger.warn('add_raw_transaction %d/%d' % (len(txins), len(txouts)))
+        # logger.debug(pprint.pformat(txins))
+        # logger.debug(pprint.pformat(txouts))
+
+        util_test.save_rawtransaction(self.rawtransactions_db, tx_id, raw_transaction)
+
+        self.update_utxo_set(txins, txouts)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def setup_logging():
+    print("")  # for --verbose output this makes sure the logs start on a newline
+    log.set_up(log.ROOT_LOGGER, verbose=True, console_logfilter=os.environ.get('COUNTERPARTY_LOGGING', None))
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_utxos(rawtransactions_db):
+    with open(util_test.CURR_DIR + '/fixtures/unspent_outputs.json', 'r') as listunspent_test_file:
+        util_test.MOCK_UTXO_SET = MockUTXOSet(json.load(listunspent_test_file), rawtransactions_db)
+
+    util_test.UNIQUE_DUMMY_TX_HASH = {}
+    util_test.TX_INDEX = 1
```

Comment:
`scope=function, autouse=True` so this is where we reset stuff between each test